src/applications/bmqtool/m_bmqtool_inpututil.cpp

```diff
@@ -257,15 +302,152 @@ void InputUtil::verifyProperties(
+    bdlb::Tokenizer           tokenizer(properties, " ");
+    bdlb::Tokenizer::iterator tokenizerIt = tokenizer.begin();
+    MessageProperty           messageProperty;
+    State                     state = NAME;
+    // Process tokens skipping open marker
+    for (++tokenizerIt; tokenizerIt != tokenizer.end(); ++tokenizerIt) {
+        bslstl::StringRef token = *tokenizerIt;
+
+        if (token == "=" || token == "]")
+            continue;
+
+        switch (state) {
+        case NAME: {
+            messageProperty.name() = token;
+            state                  = TYPE;
+        } break;  // BREAK
+        case TYPE: {
+            // Add enum prefix, remove surrounding brackets, and convert to
+            // MessagePropertyType enum
+            bsl::string typeStr = "E_";
+            typeStr.append(token.substr(1, token.size() - 2));
+            if (MessagePropertyType::fromString(&messageProperty.type(),
+                                                typeStr) != 0) {
+                if (error) {
+                    error->assign("Failed to convert MessagePropertyType");
+                }
+                return false;  // RETURN
+            }
+            state = VALUE;
+        } break;  // BREAK
+        case VALUE: {
+            // Special handling for binary value
+            if (messageProperty.type() ==
+                MessagePropertyType::Value::E_BINARY) {
+                // Check opening quote mark presense
+                if (token.at(0) != '"') {
+                    if (error) {
+                        error->assign(
+                            "Binary value opening quote mark missed");
+                    }
+                    return false;  // RETURN
+                }
+                // Join tokens by space (detokenize) inside surrounding quotes.
+                // Quote mark after end of the line is the end of hexdump.
+                mwcu::MemOutStream resultStream(' ');
+                for (++tokenizerIt; tokenizerIt != tokenizer.end();
+                     ++tokenizerIt) {
+                    token = *tokenizerIt;
+                    // Check closing quote mark after end line
+                    if (token.find("|\n\"") != bsl::string::npos) {
+                        break;  // BREAK
+                    }
+
+                    resultStream << token << ' ';
+                }
+
+                messageProperty.value() = resultStream.str();
+            }
+            else if (messageProperty.type() ==
+                     MessagePropertyType::Value::E_STRING) {
+                // Special handling for string value: it may contain spaces
+                // Join tokens by space (detokenize) inside surrounding quotes.
+                mwcu::MemOutStream resultStream;
+                for (; tokenizerIt != tokenizer.end(); ++tokenizerIt) {
+                    token = *tokenizerIt;
+                    resultStream << token;
+                    if (token.back() == '"') {
```

Comment:
According to [Tokenizer doc](https://github.com/bloomberg/bde/blob/main/groups/bdl/bdlb/bdlb_tokenizer.h#L30)
```
A *soft* *delimiter* is a maximal (non-empty) sequence of soft-delimiter
characters.  Soft delimiters, typically whitespace characters, are used to
separate (rather than terminate) tokens, and thus never result in an empty
token.
```